controllers/configmap_controller.go

```diff
@@ -288,5 +309,25 @@ func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // isValidConfigMap returns true if the ConfigMap object is the InstanceConfigMap
 func (r *ConfigMapReconciler) isValidConfigMap(o client.Object) bool {
-	return o.GetNamespace() == r.watchNamespace && o.GetName() == wiparser.InstanceConfigMap
+	return o.GetNamespace() == r.watchNamespace &&
+		(o.GetName() == wiparser.InstanceConfigMap || o.GetName() == servicescm.ServicesConfigMapName)
+}
+
+// createServicesConfigMap creates a valid ServicesConfigMap and returns it
+func (r *ConfigMapReconciler) createServicesConfigMap(ctx context.Context, namespacedName kubeTypes.NamespacedName) (
+	*core.ConfigMap, error) {
+	windowsServices, err := servicescm.Generate(namespacedName.Name, namespacedName.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = r.client.Create(ctx, windowsServices); err != nil {
```

Comment:
Won't this fail if the `ConfigMap` already exists say on a Pod restart?

Reply:
This is only called if the ConfigMap does not already exist (see Reconcile method above)

Reply:
The comment was not clear in the Reconcile function. I have called that out.

Reply:
Fixed up that comment